Composer/packages/server/src/externalContentProvider/azureBotServiceProvider.ts

```diff
@@ -0,0 +1,103 @@
+import { IContentProviderMetadata, ExternalContentProvider } from './externalContentProvider';
+// import { authService } from '../services/auth/auth';
+// import logger from '../logger';
+import { join } from 'path';
+import { remove, ensureDirSync } from 'fs-extra';
+// import { urlencoded } from 'body-parser';
+
+function prettyPrintError(err: string | Error): string {
+  if (typeof err === 'string') {
+    return err;
+  }
+  if (err && err.message) {
+    return err.message;
+  }
+  return '';
+}
```

Comment:
This exact same function exists in `powerVirtualAgentsProvider.ts`. Extract it out into a separate file so it can be reused and we don't have duplicate function definitions.